### PR TITLE
EZP-30170: [2.5LTS] Drop Varnish 4 + get rid of duplicated VCL in meta repo

### PR DESCRIPTION
**Pull request by andrerom**
Story: https://jira.ez.no/browse/EZP-30170

Additional awareness / review ping:
- @vidarl  _Demo Cloud / Platform.sh impact?_
- @Plopix  _eZ Launchpad impact?_

Todo:
- [x] Look into docker failure for behat

### CODE REVIEW

**Comment by andrerom** [comment]
@Plopix Will this change be a problem form eZ Launchpad? _(aka do you need VCL file before composer install / create-project has been run)_

@alongosz / @mnocon Same question for you, would it make sense to change setup_ezplatform.sh back to somehow run composer install before spinning up all containers?
And before you answer that, be aware we expect just that in solr contianer already: https://github.com/ezsystems/ezplatform/blob/master/doc/docker/Dockerfile-solr#L4 ;)

**Comment by mnocon** [comment]
Sadly I don't see any other clean way, given that the file is in another repository.

**Comment by vidarl** [comment]
Should be no problem for Demo Cloud / Platform.sh impact.

And since we run `composer install` before the varnish container is builded, there should be no issues getting the .vcl from vendor/

**Comment by alongosz** [comment]
> @alongosz / @mnocon Same question for you, would it make sense to change setup_ezplatform.sh back to somehow run composer install before spinning up all containers?
> And before you answer that, be aware we expect just that in solr contianer already: https://github.com/ezsystems/ezplatform/blob/master/doc/docker/Dockerfile-solr#L4 ;)

When/where exactly is this Dockerfile used? Maybe we could split the process into app (where `composer install happens`) + others after? `composer install` was moved to be done inside `app` container because of a couple of reasons:
- it sets proper permissions for environment inside docker container
- it executes composer taking into account docker environment (this might be of no impact if Travis Trusty env is in sync with what docker container offers, but might lead to problems in the future).

I'm not in the loop when it comes to the actual issue, but if this is related to eZ Launchpad, then maybe it calls for a separate setup script? `setup_ezplatform.sh` is done to support Travis CI only.

**Comment by andrerom** [comment]
> Maybe we could split the process into app (where `composer install happens`) + others after?

This is what it did before _(first build stage to pull in packages)_. This is also what launchpad and sales demo does, and what is required for solr contianer as well already. There are many ways to do it, either just using travis php, or run php container first, before orchestrating the whole setup in second phase.

**Comment by andrerom** [comment]
As removing the files depends on QA, I'll just deprecate it for now.

So smaller PR here ready for review @lserwatka @mnocon

File: doc/varnish/varnish.md [hunk 1]
```diff
@@ -3,13 +3,12 @@ eZ Platform Varnish configuration
...
-* A working Varnish 3 or Varnish 4 setup.
+* A working Varnish 5.1 or higher _(6.0 is a LTS, so the recommended version we test against)_.
+* [Varnish xkey module](https://github.com/varnish/varnish-modules/)
 
 Recommended VCL base files
 --------------------------
-For Varnish to work properly with eZ, you'll need to use the following VCL as starting point:
-
-* [eZ Platform 1.7+ with Varnish 4.x/5.x with xkey VMOD](vcl/varnish4_xkey.vcl)
+Provided VCL for eZ can be found in [vendor/ezsystems/ezplatform-http-cache/docs/varnish](https://github.com/ezsystems/ezplatform-http-cache/tree/0.8/docs/varnish).
```
Comment (mnocon) [inline doc/varnish/varnish.md]: Is it ok to link to 0.8 here? This VCL still supports 4.0 and needs manual changes (for example: https://github.com/ezsystems/ezplatform-http-cache/blob/master/docs/varnish/vcl/varnish4.vcl#L268) and we won't be changing it in 0.8 branch I guess.

I'd rather link to a file that is tailored for 5.1.

Comment (andrerom) [reply doc/varnish/varnish.md]: See updates here and in https://github.com/ezsystems/ezplatform-http-cache/pull/77

**Comment by andrerom** [comment]
@mnocon Did you test the slightly new VCL?

**Pull request merged**